File: api_service/backend/internal/handler/response.go
package handler

import (
	"encoding/json"
	"net/http"
)

// APIResponse is the standardized JSON response structure.
type APIResponse struct {
	Code    int         `json:"code"`
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

// RespondWithJSON is a helper to write a standard JSON response.
func RespondWithJSON(w http.ResponseWriter, code int, message string, payload interface{}) {
	response := APIResponse{
		Code:    code,
		Message: message,
		Data:    payload,
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(code)
	json.NewEncoder(w).Encode(response)
}

// RespondWithError is a helper for error responses.
func RespondWithError(w http.ResponseWriter, code int, message string) {
	RespondWithJSON(w, code, message, nil)
}

File: api_service/cf/src/response.ts
import { Context } from 'hono';

/**
 * Standardized API response structure, mirroring the Go backend.
 */
interface APIResponse {
  code: number;
  message: string;
  data?: any;
}

/**
 * A helper function to create standardized JSON responses.
 * @param c Hono context
 * @param statusCode HTTP status code
 * @param message A descriptive message
 * @param data The payload to be sent (optional)
 * @returns A Response object.
 */
export function respond(c: Context, statusCode: number, message: string, data?: any) {
  const response: APIResponse = {
    code: statusCode,
    message,
  };
  if (data !== undefined) {
    response.data = data;
  }
  return c.json(response, statusCode as any);
}
